src/components/withNavigationContext.js

```diff
@@ -0,0 +1,68 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {NavigationContext} from '@react-navigation/native';
+import getComponentDisplayName from '../libs/getComponentDisplayName';
+
+const navigationContextPropTypes = {
+    // Whether this screen is focused
+    isScreenFocused: PropTypes.bool.isRequired,
+};
+
+export default function (WrappedComponent) {
+    class withNavigationContext extends Component {
+        constructor(props) {
+            super(props);
+            this.state = {
+                isScreenFocused: false,
+            };
+        }
+
+        componentDidMount() {
+            this.unsubscribeFocus = this.context.addListener('focus', () => this.setState({
+                isScreenFocused: true,
+            }));
+            this.unsubscribeBlur = this.context.addListener('blur', () => this.setState({
+                isScreenFocused: false,
+            }));
+        }
+
+        componentDidUpdate() {
+            const valueToReturn = this.context.isFocused();
```

Comment:
is there a more descriptive variable name we can use for this `const`? I'm unsure what "value" is being stored here. 

If not, can we add a small comment above it describing what value we expect to get.

Reply:
what about `isCurrentlyFocused`?

Reply:
yup, that's great (if what's being returned is a `bool` vs. a number of some kind)